File: app/src/main/java/com/glitchhollow/Rect2D.java
package com.glitchhollow;

public class Rect2D {

    public float x, y, width, height;

    public Rect2D(float x, float y, float width, float height) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }

    public boolean intersects(Rect2D other) {
        return x < other.x + other.width
            && x + width > other.x
            && y < other.y + other.height
            && y + height > other.y;
    }

    public float right()  { return x + width; }
    public float bottom() { return y + height; }
    public float centerX(){ return x + width  / 2f; }
    public float centerY(){ return y + height / 2f; }

    public void set(float x, float y, float w, float h) {
        this.x = x; this.y = y;
        this.width = w; this.height = h;
    }
}

File: app/src/main/java/com/glitchhollow/core/Rect2D.kt
package com.glitchhollow.core

/** Axis-aligned bounding box for collision detection. Pure math, no Android deps. */
data class Rect2D(var x: Float, var y: Float, var w: Float, var h: Float) {

    fun set(x: Float, y: Float, w: Float, h: Float) {
        this.x = x; this.y = y; this.w = w; this.h = h
    }

    fun intersects(other: Rect2D): Boolean =
        x < other.x + other.w && x + w > other.x &&
        y < other.y + other.h && y + h > other.y

    val right  get() = x + w
    val bottom get() = y + h
}